packages/cinterop/src/nativeDarwin/kotlin/io/realm/kotlin/internal/interop/RealmInterop.kt

```diff
@@ -2225,6 +2227,34 @@ actual object RealmInterop {
         )
     }
 
+    actual fun realm_sync_session_register_progress_notifier(
+        syncSession: RealmSyncSessionPointer,
+        direction: ProgressDirection,
+        isStreaming: Boolean,
+        callback: ProgressCallback,
+    ): ULong {
+        return realm_wrapper.realm_sync_session_register_progress_notifier(
+            syncSession.cptr(),
+            direction.nativeValue,
+            isStreaming,
+            staticCFunction<COpaquePointer?, ULong, ULong, Unit> { userData, a, b ->
+                safeUserData<ProgressCallback>(userData).run {
+                    onChange(a.toLong(), b.toLong())
+                }
+            },
+            StableRef.create(callback).asCPointer(),
+            staticCFunction { userdata ->
+                disposeUserData<ProgressCallback>(userdata)
+            }
+        )
+    }
+    actual fun realm_sync_session_unregister_progress_notifier(
+        syncSession: RealmSyncSessionPointer,
+        token: ULong
```

Comment:
We could use an inline class to wrap the token.

Reply:
Good idea, but I am awaiting the implementation from core that wraps the token in a `realm_release`'able object, so the token as an integer will go away soon 🤞